src/libraries/Common/src/System/Security/Cryptography/MLDsa.cs

```diff
@@ -1189,16 +1461,18 @@ protected virtual void Dispose(bool disposing)
 
         private AsnWriter ExportSubjectPublicKeyInfoCore()
         {
-            ThrowIfDisposed();
-
-            byte[] rented = CryptoPool.Rent(Algorithm.PublicKeySizeInBytes);
-
+            int publicKeySizeInBytes = Algorithm.PublicKeySizeInBytes;
+            byte[] rented = CryptoPool.Rent(publicKeySizeInBytes);
             try
             {
-                Span<byte> keySpan = rented.AsSpan(0, Algorithm.PublicKeySizeInBytes);
-                ExportMLDsaPublicKey(keySpan);
+                Span<byte> publicKey = rented.AsSpan(0, publicKeySizeInBytes);
+                ExportMLDsaPublicKeyCore(publicKey);
 
-                AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
+                // The ASN.1 overhead of a SubjectPublicKeyInfo encoding a public key is 22 bytes.
```

Comment:
This was computed by comparing `openssl genpkey -algorithm ML-DSA-87 | openssl pkey -pubout -outform DER | wc` with the public key size for the algo.